### ISSUE TITLE
r3 does not get launched

### ISSUE THREAD

**Issue by MarkoSchuetz** [metaeducation/rebol-server#8]
Whenever I launch rebol-server I get an error message: 
 ```
Cannot run program "/data/user/0/giuliolunati.rebolserver/files/system/r3" (in directory "/data/user/0/giuliolunati.rebolserver/files"): error=20, Not a directory
```

**Comment by giuliolunati** [comment]
Sorry, I'm working on that

Il dom 27 dic 2020, 18:56 Marko Schütz-Schmuck <notifications@github.com>
ha scritto:

**Comment by giuliolunati** [comment]
Changed prebuilt binary source, should work now.
NOTE: please remove completely the app before reinstall

**Comment by MarkoSchuetz** [comment]
I uninstalled the APK on the phone and did a fresh git clone & build. Now I get
`Attempt to invoke virtual method 'int java.lang.String.indexOf(java.lang.String)' on a null object reference`

**Comment by hostilefork** [comment]
I'm trying to add [building and testing of the rebol-server.apk](https://github.com/metaeducation/rebol-server/issues/9) to the GitHub CI build for Android, so I made a rebol-server.apk.

I've done this only once before, and when I did, I did it in Termux on an Android phone [as in @giuliolunati's conference video](https://youtu.be/r5kccBehMMg?t=500).  However, since that's not pleasant to try and use a debug bridge in an emulator to automate, I did a build on the host.  Here is the resulting APK:

[http://hostilefork.com/media/shared/android/rebol-server.apk](http://hostilefork.com/media/shared/android/rebol-server.apk)

It works for me on my Moto G6, as well as in an API Level 30 Emulator that comes in a relatively recent Android Studio.  However, it gives a dialog box complaining that it was built for an older version of Android and to ask for a new version.

**Comment by hostilefork** [comment]
I'm pretty sure this was the result of not having --recurse-submodules in the enlistment.

The GitHub Action for building the server just goes ahead and [overwrites the committed directories with the latest versions of rebol-httpd and replpad-js](https://github.com/metaeducation/ren-c/blob/176aa15eaa8d9746948f5533a801801bb1bab75c/.github/workflows/android-build.yml#L354).

If the latest binary of the interpreter is being pulled from s3, then what probably makes sense is to check to see if you have a rebol-httpd and replpad-js enlistment and just use it if you do, otherwise get the latest.  This is how the interpreter is handled.

Will make that a separate issue and close this one.
